File: src/main/java/com/waracle/demo/service/CakeService.java
package com.waracle.demo.service;

import java.util.List;

import com.waracle.demo.dto.CakeDto;

/**
 * Created by Nilesh
 */
public interface CakeService {
    void saveCake(CakeDto cakeDto);
    List<CakeDto> getAllCakes();
}

File: src/main/resources/static/js/app/CakeService.js
'use strict'

angular.module('demo.services', []).factory('CakeService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getAllCakes = function() {
				return $http.get(CONSTANTS.getAllCakes);
			}
			service.saveCake = function(cakeDto) {
				return $http.post(CONSTANTS.saveCake, cakeDto);
			}
			return service;
		} ]);